roadmap 2025 (#5906)
qdrant/qdrant: docs/roadmap/README.md

# Qdrant 2025 Roadmap

Hi!
This document is our plan for Qdrant development in 2025.
Previous year roadmap is available here:

* [Roadmap 2024](roadmap-2024.md)
* [Roadmap 2023](roadmap-2023.md)
* [Roadmap 2022](roadmap-2022.md)

Goals of the release:

* **Maintain easy upgrades** - we plan to keep backward compatibility for at least one minor version back (this stays the same in 2025)
  * We plan to maintain forward client compatibility for multiple minor versions. Version incompatibility will be announced in the release notes. Also, we will track version compatibility between server and client automatically, and notify you if something is out of sync.

>   * That means that you can upgrade Qdrant without any downtime and without any changes in your client code within one minor version.

Reviewer: ```suggestion
  * That means that you can upgrade Qdrant without any downtime and without any changes in your client code within one or more minor versions.
```

  * Storage should be compatible between any two consequent versions, so you can upgrade Qdrant with automatic data migration between consecutive versions.
  * We plan to eventually migrate away from RocksDB, so at some point we might introduce data migration into the upgrade process.

> * **Continue to improve scalability** - Qdrant should be a go-to solution if you are running a billion-scale vector search. We already progressed a lot in this are, but there are a few places we can improve.

Reviewer: ```suggestion
* **Continue to improve scalability** - Qdrant should be a go-to solution if you are running a billion-scale vector search. We already progressed a lot in this area, but there are still a few places where we can improve.
```

* **Focus on vector similarity** - we believe that vector similarity is not only about search. Vector similarity reveals the structure of your data, even if you don't have specific search queries. We plan to introduce more functions, demos and tutorials targeting this use-case.

> * **End-to-end processing** - We plan to introduce tighter integration with embedding provides, so close gap between local development and production deployment.

Reviewer 2: ```suggestion
* **End-to-end processing** - We plan to introduce tighter integration with embedding providers, so close gap between local development and production deployment.
```

> * **End-to-end processing** - We plan to introduce tighter integration with embedding provides, so close gap between local development and production deployment.

Reviewer 2: ```suggestion
* **End-to-end processing** - We plan to introduce tighter integration with embedding provides, to close the gap between local development and production deployment.
```
